saleor/graphql/giftcard/tests/test_gift_card.py

```diff
@@ -179,6 +179,58 @@ def test_query_own_gift_cards(user_api_client, gift_card, gift_card_created_by_s
     assert data["totalCount"] == 1
 
 
+def test_query_gift_card_by_app_with_permissions(
+    app_api_client, gift_card, permission_manage_gift_card, permission_manage_users
+):
+    query = """
+    query giftCard($id: ID!) {
+        giftCard(id: $id){
+            id
+            displayCode
+            user {
+                email
+            }
+        }
+    }
+    """
+    gift_card_id = graphene.Node.to_global_id("GiftCard", gift_card.pk)
+    variables = {"id": gift_card_id}
+
+    # Query should fail without manage_users permission.
+    response = app_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_gift_card]
+    )
+    assert_no_permission(response)
```

Comment:
Would be a good idea to separate this into separate test and keep no_permission in one and with the permission in other. Otherwise if that test fails I have to analyze logic in it to actually see what happened.

Reply:
We decided to left it as it is, for now.